src/utils/user/fetchGroups.ts

```diff
@@ -21,15 +21,26 @@ export const fetchGroups = async () => {
     description,
   } = config.get('user.groups.list') as GroupListSettings;
 
+  if (!apiConfigurationID) {
+    return [];
+  }
+
   const apiConfiguration = await ApiConfiguration.findById(apiConfigurationID);
+
   let data: any;
 
   // Switch on all available authTypes
   if (
     apiConfiguration.authType === authType.serviceToService ||
-    apiConfiguration.authType === authType.userToService
+    apiConfiguration.authType === authType.userToService ||
+    apiConfiguration.authType === authType.token
   ) {
     const token: string = await getToken(apiConfiguration);
+
+    if (!token) {
+      return [];
+    }
+
```

Comment:
This is not safe enough, we have to see how to prevent groups fetching from executing and warn the user that it cannot be done with U2S

Reply:
Done by adding a warning, as the fetchGroups api config is harcoded in config file.